x-pack/plugins/security_solution/server/usage/endpoints/fleet_saved_objects.ts

```diff
@@ -5,38 +5,40 @@
  * 2.0.
  */
 
-import { ISavedObjectsRepository } from 'src/core/server';
+import { ElasticsearchClient, SavedObjectsClientContract } from 'src/core/server';
+import { AgentService } from '../../../../fleet/server';
 import { AgentEventSOAttributes } from './../../../../fleet/common/types/models/agent';
 import {
   AGENT_SAVED_OBJECT_TYPE,
   AGENT_EVENT_SAVED_OBJECT_TYPE,
 } from './../../../../fleet/common/constants/agent';
-import { Agent, defaultPackages as FleetDefaultPackages } from '../../../../fleet/common';
+import { defaultPackages as FleetDefaultPackages } from '../../../../fleet/common';
 
 export const FLEET_ENDPOINT_PACKAGE_CONSTANT = FleetDefaultPackages.Endpoint;
 
-export const getFleetSavedObjectsMetadata = async (savedObjectsClient: ISavedObjectsRepository) =>
-  savedObjectsClient.find<Agent>({
-    // Get up to 10000 agents with endpoint installed
-    type: AGENT_SAVED_OBJECT_TYPE,
-    fields: [
-      'packages',
-      'last_checkin',
-      'local_metadata.agent.id',
-      'local_metadata.host.id',
-      'local_metadata.host.name',
-      'local_metadata.host.hostname',
-      'local_metadata.elastic.agent.id',
-      'local_metadata.os',
-    ],
-    filter: `${AGENT_SAVED_OBJECT_TYPE}.attributes.packages: ${FLEET_ENDPOINT_PACKAGE_CONSTANT}`,
+export const getFleetSavedObjectsMetadata = async (
+  savedObjectsClient: SavedObjectsClientContract,
+  agentService: AgentService | undefined,
+  esClient: ElasticsearchClient
+) => {
+  const agentData = await agentService?.listAgents(savedObjectsClient, esClient, {
```

Comment:
I just merged the change where we remove the support for saved object, I think this method is probably not accepting a savedObjectClient any more

Reply:
Thanks for the note. I'll update this accordingly!